pkg/kube/util/meltdown/meltdown.go

```diff
@@ -0,0 +1,54 @@
+package meltdown
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"code.cloudfoundry.org/cf-operator/pkg/kube/apis"
+)
+
+// AnnotationLastReconcile is the key name of the timestamp for the last
+// successful reconcile
+var AnnotationLastReconcile = fmt.Sprintf("%s/last-reconcile", apis.GroupName)
+
+// Window represents a time window starting at start and ending at duration
+type Window struct {
+	Start    time.Time
+	Duration time.Duration
+}
+
+// Contains returns true if the given time is in the specified time window
+func (w Window) Contains(now time.Time) bool {
+	if now.Before(w.Start) {
+		return false
+	}
+	windowEnd := w.Start.Add(w.Duration)
+	return now.Before(windowEnd)
+}
+
+// NewWindow returns a new time window starting at lastReconcile, ending after duration
+func NewWindow(duration time.Duration, lastReconcile *metav1.Time) Window {
+	if lastReconcile != nil {
+		start := lastReconcile.Time
+		return Window{Start: start, Duration: duration}
+	}
+	return Window{}
```

Comment:
Is it we expect that return empty window, even nil Duration? do we have any case like that?

Reply:
If you have no last reconcile timestamp this will return the year 0 and you will always be out of the duration.
Of course, this will break if your window duration is a few thousand years, but otherwise we're fine.